controller/ingress_entry.go

```diff
@@ -6,9 +6,24 @@ import (
 	"time"
 )
 
-// IngressUpdate data
-type IngressUpdate struct {
-	Entries []IngressEntry
+// IngressEntries type
+type IngressEntries []IngressEntry
+
+// allows us to sort ingress entries.
+func (entries IngressEntries) Len() int {
+	return len(entries)
+}
+
+// allows us to sort ingress entries.
+func (entries IngressEntries) Less(i, j int) bool {
+	iEntry := fmt.Sprintf("%s:%s:%s:%s", entries[i].Namespace, entries[i].Name, entries[i].Host, entries[i].Path)
+	jEntry := fmt.Sprintf("%s:%s:%s:%s", entries[j].Namespace, entries[j].Name, entries[j].Host, entries[j].Path)
```

Comment:
Use `strings.Join`. Also I'm not sure how safe it is to use ":" as the separator. It would be faster/more correct to compare each item separately, but a lot more code.